src/summary.lua

```diff
@@ -1,34 +1,55 @@
 SetHeader("Access-Control-Allow-Origin", "*")
 
+if GetMethod() ~= 'GET' and GetMethod() ~= 'HEAD' then
+    Log(kLogWarn, "got %s request from %s" % {GetMethod(), FormatIp(GetRemoteAddr() or "0.0.0.0")})
+    ServeError(405)
+    SetHeader('Allow', 'GET, HEAD')
+    return
+end
+
 local smallest, biggest
 local cidr = GetParam("subnet")
 if cidr ~= nil and cidr ~= "" then
     local full, ip, mask = re.search([[^([0-9.]{7,15})/([0-9]{1,2})$]], cidr)
     if full == nil then
-        return ServeError(400, "Invalid CIDR")
+        SetStatus(400, "need subnet=x.x.x.x/y parameter")
+        Write("need subnet=x.x.x.x/y parameter")
+        return
     end
     ip = ParseIp(ip)
     if ip < 0 then
-        return ServeError(400, "Invalid CIDR")
+        SetStatus(400, "invalid ip address")
+        Write("invalid ip address")
+        return
     end
     mask = tonumber(mask)
-    if mask > 8 then
-        return ServeError(400, "Mask >8 not supported")
+    if mask < 24 or mask > 31 then
```

Comment:
This is definitely wrong - I want to allow only highly broad summaries, but this only allows really narrow ones. I'd rather just not provide the endpoint. See #5 and #6